lib/Routing/Router.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Routing;
+
+use Doctrine\Website\Site;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGenerator;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\Matcher\UrlMatcher;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use function explode;
+use function parse_url;
+
+class Router
+{
+    /** @var RouteCollection */
+    private $routes;
+
+    /** @var RequestContext */
+    private $context;
+
+    /**
+     * @param mixed[] $routes
+     */
+    public function __construct(array $routes, Site $site)
+    {
+        $this->routes = new RouteCollection();
+
+        $url = parse_url($site->getUrl());
+
+        $this->context = new RequestContext(
+            '',
+            'GET',
+            $url['host'],
+            $url['scheme'],
+            $url['port'],
```

Comment:
Can you make the port optional? This new version of the website doesn't allow `doctrine.website.url` without a port.

Notice: Undefined index: port in lib/Routing/Router.php on line 40